python/python_fastapi_server/routers/edr.py

```diff
@@ -0,0 +1,876 @@
+                        "dataset":
+                        dataset,
+                        "name":
+                        edr_collection.attrib.get("name"),
+                        "service":
+                        "http://localhost:8000/wms",
+                        "time_interval":
+                        edr_collection.attrib.get("time_interval"),
+                        "parameters": [
+                            inst.strip()
+                            for inst in edr_collection.text.strip().split(",")
+                        ],
+                        "vertical_name": edr_collection.attrib.get("vertical_name"),
+                        "base_url": f"http://localhost:8080/edr/collections/{edr_collection.attrib.get('name')}"
+                    }
+        except ParseError:
+            pass
+    return edr_collections
+
+global_edr_collections=init_edr_collections()
+
+def get_edr_collections():
+    return global_edr_collections
+
+def get_point_value(
+    edr_collectioninfo: dict,
+    instance: str,
+    coords: list[float],
+    parameters: list[str],
+    datetime_par: str,
+    z_par: str = None,
+    custom_dims: str=None,
+):
+    dataset=edr_collectioninfo['dataset']
+    urlrequest = (
+        f"SERVICE=WMS&VERSION=1.3.0&REQUEST=GetPointValue&CRS=EPSG:4326"
+        f"&DATASET={dataset}&QUERY_LAYERS={','.join(parameters)}"
+        f"&X={coords[0]}&Y={coords[1]}&INFO_FORMAT=application/json")
+    if datetime_par:
+        urlrequest += f"&TIME={datetime_par}"
+
+    if instance:
+        urlrequest += f"&DIM_reference_time={instance_to_iso(instance)}"
+    if z_par:
+        if "vertical_name" in edr_collectioninfo and edr_collectioninfo["vertical_name"]!="ELEVATION":
+            urlrequest += f"&DIM_{edr_collectioninfo['vertical_name']}={z_par}"
+        else:
+            urlrequest += f"&ELEVATION={z_par}"
+    if custom_dims:
+        urlrequest += custom_dims
+
+    logger.info("URL: %s", urlrequest)
+    status, response = call_adaguc(url=urlrequest.encode("UTF-8"))
+    logger.info("status: %d", status)
+    if status == 0:
+        return response.getvalue()
+    return None
+
+
+@edrApiApp.get(
+    "/collections/{collection_name}/position",
+ #   response_model=CovJSONResponse,
+    response_model_exclude_none=True,
+)
+@edrApiApp.get(
+    "/collections/{collection_name}/instances/{instance}/position",
+#    response_model=CovJSONResponse,
+    response_model_exclude_none=True,
+)
+async def get_collection_position(
+        collection_name: str,
+        request: Request,
+        coords: str,
+        instance:  Union[str, None],
+        datetime_par: str = Query(default=None, alias="datetime"),
+        parameter_name: str = Query(alias="parameter-name"),
+        z_par: str=Query(alias="z", default=None),
+        crs: str=Query(default=None),
+        requested_format: str=Query(default=None, alias="f")
+)->CovJSONResponse:
```

Comment:
This is what I had in mind:

```suggestion
@edrApiApp.get(
    "/collections/{collection_name}/position",
    response_model=Coverage,
    response_class=CovJSONResponse,
    response_model_exclude_none=True,
)
@edrApiApp.get(
    "/collections/{collection_name}/instances/{instance}/position",
    response_model=Coverage,
    response_class=CovJSONResponse,
    response_model_exclude_none=True,
)
async def get_collection_position(
        collection_name: str,
        request: Request,
        coords: str,
        instance:  Union[str, None],
        datetime_par: str = Query(default=None, alias="datetime"),
        parameter_name: str = Query(alias="parameter-name"),
        z_par: str=Query(alias="z", default=None),
        crs: str=Query(default=None),
        requested_format: str=Query(default=None, alias="f")
) -> Coverage:
```